quickwit/quickwit-indexing/src/split_store/indexing_split_cache.rs

```diff
@@ -170,14 +227,39 @@ impl InnerLocalSplitStore {
         split_id: Ulid,
         to_folder: &Path,
     ) -> StorageResult<Option<PathBuf>> {
-        let Some(split_folder) = self.split_folders.remove(&split_id) else {
+        if !self.split_registry.remove(split_id) {
```

Comment:
i'm not sure about the semantic on the error path. We forget we have the split, remove it from quota, but leave it on disk.
I think checking if the split is known, trying to move it, and then remove from the registry might be better, but am not 100% sure.